RecoTauTag/RecoTau/plugins/PFRecoTauChargedHadronFromTrackPlugin.cc

```diff
@@ -143,13 +143,14 @@ PFRecoTauChargedHadronFromTrackPlugin::return_type PFRecoTauChargedHadronFromTra
   evt.getByToken(Tracks_token, tracks);
 
   qcuts_->setPV(vertexAssociator_.associatedVertex(jet));
-
+  float jEta=jet.eta();
+  float jPhi=jet.phi();
   size_t numTracks = tracks->size();
   for ( size_t iTrack = 0; iTrack < numTracks; ++iTrack ) {
     reco::TrackRef track(tracks, iTrack);
 
     // consider tracks in vicinity of tau-jet candidate only
-    double dR = deltaR(track->eta(), track->phi(), jet.eta(), jet.phi());
+    double dR = deltaR(track->eta(), track->phi(), jEta,jPhi);
     double dRmatch = dRcone_;
     if ( dRconeLimitedToJetArea_ ) {
```

Comment:
no, maybe we should compile one and provide some clever algorithm to perform NN searches in eta-phi.
The problem is, as with any combinatorics, that the optimal algorithm depends on the complexity of the problem
(see fastjet).
In principle one could use fastjet each time deltaR is used...  (or PF algos)